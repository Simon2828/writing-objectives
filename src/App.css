html {
  box-sizing: border-box;
  font-size: 16px;
  font-family: sans-serif;
}

*, *:before, *:after {
  box-sizing: inherit;
}

body {
  margin: 0;
  padding: 0;
  font-weight: normal;
}

ol, ul {
  list-style: none;
  padding: 0;
}

img {
  max-width: 100%;
  height: auto;
}
