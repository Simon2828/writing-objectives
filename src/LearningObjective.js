import React, {useContext} from 'react';
import { LearningObjectivesContext } from './LearningObjectivesContext';

const LearningObjective = (state) => {

  const [learningObjectives] = useContext(LearningObjectivesContext);
  const learningObjective = state.location.state.objective.item
  const learningObjectiveRefIndex = state.location.state.objective.refIndex
  const loId = learningObjectives.learningObjectives.allIds[learningObjectiveRefIndex];

  const stepsToSuccess = learningObjectives.learningObjectives.byId[loId].stepsToSuccess.map((step, i) => {
    return <li key={i}>{learningObjectives.stepsToSuccess.byId[step].stepToSuccess}</li>});

  return (
    <>
    <h4>{learningObjective}</h4>
    <ul>{stepsToSuccess}</ul>
    </>
  );
  }

export default LearningObjective;