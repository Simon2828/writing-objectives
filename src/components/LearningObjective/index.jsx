import React, { useContext } from "react";
import { Link } from "react-router-dom";
import StepsToSuccess from "../StepsToSuccess";
import StyledStepToSuccess from "../StepToSuccess";
import { LearningObjectivesContext } from "../../LearningObjectivesContext";
import styled from "styled-components";
import EditLearningObjective from "../EditLearningObjective";

const FlexWrapper = styled.ul`
  display: flex;
  margin: 0.8rem auto 5rem;
  flex-direction: column;
  width: 90%;
  @media only screen and (min-width: 600px) {
    width: 80%;
    > a {
      display: flex;
      align-items: center;
      justify-content: center;
      text-decoration: none;
      color: black;
    }
  }
`;

const StyledLearningObjective = styled.h2`
  border-bottom: rgb(228, 230, 232) 1px solid;
  width: 100%;
`;

const StyledAllDone = styled.div`
  margin: 2rem;
  font-size: 1.5rem;
`;

const StyledButton = styled.button`
  color: palevioletred;
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  border: 2px solid palevioletred;
  border-radius: 3px;
  background-color: white;
  cursor: pointer;
`;

const LearningObjective = (state) => {
console.log("ðŸš€ ~ file: index.jsx ~ line 48 ~ LearningObjective ~ state", state)

  const [learningObjectives, setState] = useContext(LearningObjectivesContext);
  const learningObjective = state.location.state.objective.item;
  const learningObjectiveRefIndex = state.location.state.objective.refIndex;
  // is there a better way to update state of checkbox below?
  const handleCheckClick = (id) => {
    setState((learningObjectives) => ({
      ...learningObjectives,
      stepsToSuccess: {
        ...learningObjectives.stepsToSuccess,
        byId: {
          ...learningObjectives.stepsToSuccess.byId,
          [id]: {
            achieved: !learningObjectives.stepsToSuccess.byId[id].achieved,
            id: id,
            stepToSuccess:
              learningObjectives.stepsToSuccess.byId[id].stepToSuccess,
          },
        },
      },
    }));
  };
  const loId =
    learningObjectives.learningObjectives.allIds[learningObjectiveRefIndex];

  const allStepsToSuccessCompleted = () => {
    return learningObjectives.learningObjectives.byId[
      loId
    ].stepsToSuccess.every((stepId) => {
      return learningObjectives.stepsToSuccess.byId[stepId].achieved;
    });
  };

  const stepsToSuccess = learningObjectives.learningObjectives.byId[
    loId
  ].stepsToSuccess.map((step, i) => {
    return (
      <StyledStepToSuccess
        key={i}
        stepToSuccess={
          learningObjectives.stepsToSuccess.byId[step].stepToSuccess
        }
        id={step}
        loId={loId}
        onClick={handleCheckClick}
        achieved={learningObjectives.stepsToSuccess.byId[step].achieved}
      />
    );
  });

  const stepsToSuccessToEdit = learningObjectives.learningObjectives.byId[
    loId
  ].stepsToSuccess.map((step, i) => {
    return (
        <li key={i}>{learningObjectives.stepsToSuccess.byId[step].stepToSuccess}</li>
        // id={step}
    );
  });

// TODO - edit above to return steps to success to edit

  return (
    <FlexWrapper>
      <Link to={{
        pathname: "/learning-objective/edit",
        learningObjective,
        loId,
        stepsToSuccessToEdit
    }} >
        <StyledButton>Edit</StyledButton>
      </Link>
      <StyledLearningObjective>{learningObjective}</StyledLearningObjective>
      <StepsToSuccess>{stepsToSuccess}</StepsToSuccess>
      {allStepsToSuccessCompleted() && (
        <Link to="/well-done">
          <StyledAllDone>All done?</StyledAllDone>
          <StyledButton>Yes!</StyledButton>
        </Link>
      )}
    </FlexWrapper>
  );
};

export default LearningObjective;
