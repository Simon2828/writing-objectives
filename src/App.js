import React, { useContext, useEffect } from 'react';
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import './App.css';
import { LearningObjectivesProvider, initialData } from './LearningObjectivesContext';
import Search from './Search';
import LearningObjective from './components/LearningObjective';
import WellDone from './components/WellDone';
import StyledNavigation from './components/Navigation';
import EditLearningObjective from './components/EditLearningObjective';
import { LearningObjectivesContext } from './LearningObjectivesContext';
import Auth0ProviderWithHistory from './auth/auth0-provider-with-history';
import ProtectedRoute from './auth/protected-route';

function App() {
  // TODO: need to work out useeffect for when calling from api


  return (
    <Router>
      <Auth0ProviderWithHistory>
        <LearningObjectivesProvider>
          <StyledNavigation />
          <Switch>
            <ProtectedRoute path="/learning-objective/edit" component={EditLearningObjective} />
            <Route path="/learning-objective" component={LearningObjective} />
            <Route path="/well-done" component={WellDone} />
            {/* <Route path="/teacher" component={Login} /> */}
            {/* <Route path="/teacher/edit" component={EditLearningObjective} /> */}
            {/* <Route exact path="/" render={(props) => <Search globalStore={globalStore} {...props} /> */}
            <Route exact path="/" component={Search} />
          </Switch>
        </LearningObjectivesProvider>
      </Auth0ProviderWithHistory>

    </Router>
  );
}

export default App;
