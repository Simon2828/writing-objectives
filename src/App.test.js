import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from './App';
import LearningObjective from './LearningObjective';

// test('renders learn react link', () => {
//   const { getByText } = render(<App />);
//   const linkElement = getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

test('Learning Objective renders a learning objective and its steps to success', () => {

// arrange
  const lO = "Write an ending" //how to deal with state in tests - mock it??

  const mockState = {state: {location}}



  console.log({mockState})


// act
  const learningObjectiveComponent = renderer.create(<LearningObjective state={mockState}/>);
  let tree = learningObjectiveComponent.toJSON();

// assert
  expect(tree).toMatchSnapshot();
})

// 