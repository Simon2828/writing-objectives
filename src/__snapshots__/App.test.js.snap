// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Learning Objective renders a learning objective and its steps to success 1`] = `null`;
